src/metabase/sync_database/analyze.clj

```diff
@@ -259,3 +175,11 @@
             (log/info (u/format-color 'blue "%s Analyzed table '%s'." (u/emoji-progress-bar <> tables-count) table-name))))))
 
     (log/info (u/format-color 'blue "Analysis of %s database '%s' completed (%s)." (name driver) (:name database) (u/format-nanoseconds (- (System/nanoTime) start-time-ns))))))
+
+(defn analyze-database
+  "analyze all the tables in one database"
+  [db]
+  (analyze-data-shape-for-tables! (->> db
+                                       :id
```

Comment:
sounds like you found a bug, I'll make sure this works. I'll make sure this to works as it does on master now.